diff --git a/node_modules/chess.js/chess.js b/node_modules/chess.js/chess.js
index dd663c7..2bbee25 100644
--- a/node_modules/chess.js/chess.js
+++ b/node_modules/chess.js/chess.js
@@ -626,7 +626,7 @@ var Chess = function (fen) {
     /* check for castling if: a) we're generating all moves, or b) we're doing
      * single square move generation on the king's square
      */
-    if (piece_type === true || piece_type === KING) {
+    if ((piece_type === true || piece_type === KING) && kings[us] !== EMPTY) {
       if (!single_square || last_sq === kings[us]) {
         /* king-side castling */
         if (castling[us] & BITS.KSIDE_CASTLE) {
@@ -636,9 +636,9 @@ var Chess = function (fen) {
           if (
             board[castling_from + 1] == null &&
             board[castling_to] == null &&
-            !attacked(them, kings[us]) &&
-            !attacked(them, castling_from + 1) &&
-            !attacked(them, castling_to)
+            (!legal || (!attacked(them, kings[us]) &&
+              !attacked(them, castling_from + 1) &&
+              !attacked(them, castling_to)))
           ) {
             add_move(board, moves, kings[us], castling_to, BITS.KSIDE_CASTLE)
           }
@@ -653,9 +653,9 @@ var Chess = function (fen) {
             board[castling_from - 1] == null &&
             board[castling_from - 2] == null &&
             board[castling_from - 3] == null &&
-            !attacked(them, kings[us]) &&
-            !attacked(them, castling_from - 1) &&
-            !attacked(them, castling_to)
+            (!legal || (!attacked(them, kings[us]) &&
+                !attacked(them, castling_from - 1) &&
+                !attacked(them, castling_to)))
           ) {
             add_move(board, moves, kings[us], castling_to, BITS.QSIDE_CASTLE)
           }
@@ -964,6 +964,11 @@ var Chess = function (fen) {
       }
     }
 
+    /* remove the king if it's captured */
+    if (kings[them] === move.to) {
+        kings[them] = EMPTY;
+    }
+
     /* if big pawn move, update the en passant square */
     if (move.flags & BITS.BIG_PAWN) {
       if (turn === 'b') {
@@ -1815,6 +1820,10 @@ var Chess = function (fen) {
       return turn
     },
 
+    set_turn: function (t) {
+      turn = t;
+    },
+
     move: function (move, options) {
       /* The move function can be called with in the following parameters:
        *
@@ -1838,7 +1847,7 @@ var Chess = function (fen) {
       if (typeof move === 'string') {
         move_obj = move_from_san(move, sloppy)
       } else if (typeof move === 'object') {
-        var moves = generate_moves()
+        var moves = generate_moves(options)
 
         /* convert the pretty move object to an ugly move object */
         for (var i = 0, len = moves.length; i < len; i++) {
